### ISSUE TITLE
ClassCastException when getting entrySet from subMap

### ISSUE THREAD

**Issue by levsa** [jankotek/mapdb#814]
I have a map created as follows:

```java
   BTreeMap<String[], String> treeMap = db.treeMap("preingest")
                .keySerializer(new SerializerArrayDelta<>(Serializer.STRING))
                .valueSerializer(Serializer.STRING).createOrOpen();
```

When getting a submap and getting the entrySet as follows:
```java
  ConcurrentNavigableMap<String[], String> subMap = treeMap.
      subMap(new String[] { "key"}, new String[] { "key", null});
  Set<Entry<String[],String>> entrySet = subMap.entrySet();
   for (Entry<String[], String> entry : entrySet) {
       String[] keyParts = entry.getKey();   // ClassCastException, cannot cast Object -> String
   }
```

I get a ClassCastException:

```
java.lang.ClassCastException: [Ljava.lang.Object; cannot be cast to [Ljava.lang.String;
```

**Comment by kunalsheth** [comment]
I am not sure why that is happening, but if you want a quick fix:
`String[] keyParts = Arrays.stream(entry.getKey()).toArray(String[]::new);`

**Comment by levsa** [comment]
Thanks! My current quick fix that worked (not sure why though) was to use Object[]:

```
    Set<String[]> keySet = subMap.keySet();
    for (Object[] keyParts: keySet)
    {
        ...
    }
```

**Comment by jankotek** [comment]
Could not replicate in 3.0 and 3.1 branch. I think it was recently fixed. Fix should be in next 3.0.x release
